runtime/src/network/iouring.rs

```diff
@@ -82,6 +92,12 @@ impl crate::Network for Network {
             .into_std()
             .map_err(|_| crate::Error::ConnectionFailed)?;
 
+        if let Some(tcp_nodelay) = self.tcp_nodelay {
```

Comment:
nit: I think we just allow if this fails in tokio? In any case, we should copy the comment from there.